runtime/stdlib/contracts/test.cdc

```diff
@@ -0,0 +1,236 @@
+/// Test contract is the standard library that provides testing functionality in Cadence.
+///
+pub contract Test {
+
+    /// Convenient function to fail a test.
+    /// Is equivalent to calling `assert(false)`.
+    ///
+    pub fun fail() {
+        assert(false)
+    }
+
+    /// Blockchain emulates a real network.
+    ///
+    pub struct Blockchain {
+
+        pub let backend: AnyStruct{BlockchainBackend}
+
+        init(backend: AnyStruct{BlockchainBackend}) {
+            self.backend = backend
+        }
+
+        /// Executes a script and returns the script return value and the status.
+        /// `returnValue` field of the result will be `nil` if the script failed.
+        ///
+        pub fun executeScript(_ script: String, _ arguments: [AnyStruct]): ScriptResult {
+            return self.backend.executeScript(script, arguments)
+        }
+
+        /// Creates a signer account by submitting an account creation transaction.
+        /// The transaction is paid by the service account.
+        /// The returned account can be used to sign and authorize transactions.
+        ///
+        pub fun createAccount(): Account {
+            return self.backend.createAccount()
+        }
+
+        /// Add a transaction to the current block.
+        ///
+        pub fun addTransaction(_ tx: Transaction) {
+            self.backend.addTransaction(tx)
+        }
+
+        /// Executes the next transaction in the block, if any.
+        /// Returns the result of the transaction, or nil if no transaction was scheduled.
+        ///
+        pub fun executeNextTransaction(): TransactionResult? {
+            return self.backend.executeNextTransaction()
+        }
+
+        /// Commit the current block.
+        /// Committing will fail if there are un-executed transactions in the block.
+        ///
+        pub fun commitBlock() {
+            self.backend.commitBlock()
+        }
+
+        /// Executes a given transaction and commit the current block.
+        ///
+        pub fun executeTransaction(_ transaction: Transaction): TransactionResult {
+            self.addTransaction(transaction)
+            let txResult = self.executeNextTransaction()!
```

Comment:
Is there any way the backend could return `nil` from a transaction execution (e.g. on a failed execution)? If so it makes sense to have this return an optional `TransactionResult?`. Otherwise if the only way this can ever be `nil` is if there is no transaction present then it makes sense to force here.

Reply:
yeah, it can only be nil if there are no transactions present in the block.